src/llfs/slot_writer.hpp

```diff
@@ -225,6 +267,110 @@ class TypedSlotWriter<PackedVariant<Ts...>> : public SlotWriter
     }
   };
 
+  /** \brief An append operation of one or more slots.
+   */
+  class MultiAppend
+  {
+   public:
+    /** \brief Initialize a MultiAppend operation using the passed TypedSlotWriter.
+     *
+     * This will obtain a lock on the slot writer's LogDevice::Writer mutex.
+     */
+    explicit MultiAppend(TypedSlotWriter& slot_writer) noexcept : writer_lock_{slot_writer}
+    {
+    }
+
+    MultiAppend(const MultiAppend&) = delete;
+    MultiAppend& operator=(const MultiAppend&) = delete;
+
+    //----- --- -- -  -  -   -
+
+    Status begin_atomic_range(const batt::Grant& caller_grant) noexcept
+    {
+      return this->writer_lock_.begin_atomic_range(caller_grant);
+    }
+
+    Status end_atomic_range(const batt::Grant& caller_grant) noexcept
+    {
+      return this->writer_lock_.end_atomic_range(caller_grant);
+    }
+
+    /** \brief Packs a single slot into the log device buffer.  Does not commit the slot; all slots
+     * are committed atomically when this->finalize() is called.
+     */
+    template <typename T, typename PackedT = PackedTypeFor<T>>
+    StatusOr<SlotParseWithPayload<const PackedT*>> typed_append(const batt::Grant& caller_grant,
+                                                                T&& payload)
+    {
+      // Calculate packed size of payload.
+      //
+      const usize slot_payload_size = sizeof(PackedVariant<Ts...>) + packed_sizeof(payload);
+
+      // Allocate log buffer space and write the slot header (varint).
```

Comment:
varint -> variant

Reply:
It's not a typo -- varint means "variable-length integer" in this case.